extensions-contrib/distinctcount/src/main/java/io/druid/query/aggregation/distinctcount/DistinctCountAggregator.java

```diff
@@ -42,11 +42,13 @@ public DistinctCountAggregator(
   @Override
   public void aggregate()
   {
-    IndexedInts row = selector.getRow();
-    for (int i = 0; i < row.size(); i++) {
-      int index = row.get(i);
-      mutableBitmap.add(index);
-    }
+
+    boolean countNulls = !selector.nameLookupPossibleInAdvance() || NullHandlingHelper.useDefaultValuesForNull();
```

Comment:
I think it should be computed on higher level, producing two different classes. The lambda below is capturing, i. e. it's an allocation on each iteration

Reply:
I don't see why `nameLookupPossibleInAdvance()` is relevant here. In logic below, name lookup is not done "in advance", it's done "after row is returned"

Reply:
Also, if `nameLookupPossibleInAdvance()`, index for `null` should be pre-computed and index compared in hot loop, instead of making (potentially) expensive `lookupName()`. Or, when `nameLookupPossibleInAdvance()` returns `true`, it could be guaranteed in DimensionSelector's specification, that index of `null` must be 0 in this case

Reply:
done.